### PR TITLE
[python] Remove imports from werkzeug._compat

### PR DESCRIPTION
**Pull request by Hydrophobefireman**
werkzeug._compat was removed in 2.0, this PR removes the need to import from it and inlines couple of functions 
This removes the need for pinning werkzeug ( #6272 ) and allows users to use the latest version

#### Tests

- [ ] The code changed/added as part of this PR has been covered with tests
- [ ] All tests pass locally with `yarn test-unit`

#### Code Review

- [ ] This PR has a concise title and thorough description useful to a reviewer
- [ ] Issue from task tracker has a link to this PR

### CODE REVIEW

**Review by styfle — CHANGES_REQUESTED** [review]
This is failing the first test fixtures `./packages/python/test/fixtures/00-request-path` with the following:



```
[GET] /api/wsgi.py?query=hello%20%40world

'bytes' object has no attribute 'encode': AttributeError

Traceback (most recent call last):
  File "/var/task/vc__handler__python.py", line 154, in vc_handler
    response = Response.from_app(__vc_module.app, environ)
  File "/var/task/werkzeug/wrappers/base_response.py", line 287, in from_app
    return cls(*_run_wsgi_app(app, environ, buffered))
  File "/var/task/werkzeug/wrappers/base_response.py", line 26, in _run_wsgi_app
    return _run_wsgi_app(*args)
  File "/var/task/werkzeug/test.py", line 1096, in run_wsgi_app
    app_rv = app(environ, start_response)
  File "/var/task/flask/app.py", line 2309, in __call__
    return self.wsgi_app(environ, start_response)
  File "/var/task/flask/app.py", line 2287, in wsgi_app
    ctx = self.request_context(environ)
  File "/var/task/flask/app.py", line 2204, in request_context
    return RequestContext(self, environ)
  File "/var/task/flask/ctx.py", line 284, in __init__
    self.url_adapter = app.create_url_adapter(self.request)
  File "/var/task/flask/app.py", line 2021, in create_url_adapter
    subdomain=subdomain)
  File "/var/task/werkzeug/routing.py", line 1634, in bind_to_environ
    path_info = _get_wsgi_string("PATH_INFO")
  File "/var/task/werkzeug/routing.py", line 1631, in _get_wsgi_string
    return wsgi_decoding_dance(val, self.charset)
  File "/var/task/werkzeug/_compat.py", line 188, in wsgi_decoding_dance
    return s.encode("latin1").decode(charset, errors)
AttributeError: 'bytes' object has no attribute 'encode'
```

**Comment by styfle** [comment]
Hi @Hydrophobefireman 

Thanks for the PR!

We'll need to make sure all tests are passing before this can be merged. You can run locally with the following:

```
yarn install
yarn build
cd ./packages/python
VERCEL_TOKEN=abc123 yarn test-integration-once
```

Note that this will create real deployments on the account with the supplied token.

File: packages/python/vc_init.py [hunk 1]
```diff
@@ -82,14 +82,26 @@ def vc_handler(event, context):
...
-        from werkzeug._compat import to_bytes
-        from werkzeug._compat import wsgi_encoding_dance
         from werkzeug.datastructures import Headers
         from werkzeug.wrappers import Response
 
+        string_types = (str,)
+        def to_bytes(x, charset=sys.getdefaultencoding(), errors="strict"):
+            if x is None:
+                return None
+            if isinstance(x, (bytes, bytearray, memoryview)):  # noqa
```
Comment (styfle) [inline packages/python/vc_init.py]: What is `noqa`?

Comment (Hydrophobefireman) [reply packages/python/vc_init.py]: noqa is like the python version of eslint-ignore

https://github.com/PyCQA/pycodestyle/issues/476#issuecomment-179553820

however i don't think we need it here, I'll remove it once I fix the test

**Comment by Hydrophobefireman** [comment]
the failing test now works but looks like my machine is having trouble building rest of vercel, causing other tests to not pass*

**Review by styfle — APPROVED** [review]
Looks good, thanks!

**Comment by styfle** [comment]
This is available on the Canary channel today.

You can try it out by adding the Environment Variable `FORCE_RUNTIME_TAG=canary`

**Comment by Hydrophobefireman** [comment]
can we remove the [Werkzeug pin](https://github.com/vercel/vercel/blob/main/packages/python/src/index.ts#L89) now?

**Comment by styfle** [comment]
Yeah if all the tests pass. It would be good to add another test if pinning was causing failures.

**Pull request merged**